cloud_cleaner/resources/server.py

```diff
@@ -150,22 +150,49 @@ def send_emails(self):
         smtp_name = self._config.get_arg("smtpN")
         port = self._config.get_arg("smtpP")
         conn = self._get_conn()
+        receiver = self._config.get_arg("receiver")
+        skip_name = self._config.get_arg("skip_name")
+        servers = []
+        receivers = []
+        users = []
+
         # Loop over flagged servers to send emails to the users associated with
         # them.
         for server in self.__targets:
-            user = conn.get_user_by_id(server.user_id, False)
-            # Cannot send an email to a user with no email
-            if user.email is not None:
-                skip_name = self._config.get_arg("skip_name")
-                receiver = user.email
-                message = '''{user}, \n Your server, {server} may be deleted
-                    when its age reaches {age} if you do not change the name
-                    of the server to include {skip} at the start of
-                    the name. '''
-                message = message.format(user=user.name, server=server.name,
-                                         age=self.__age, skip=skip_name)
-                with smtplib.SMTP(smtp_name, port) as email:
-                    email.sendmail(sender, receiver, message)
+            if receiver == "":
+                # We need to collect information on who to send emails to
+                user = conn.get_user_by_id(server.user_id, False)
+                # Cannot send an email to a user with no email
+                if user.email is not None:
+                    users.append(user.name)
+                    receivers.append(user.email)
+                    servers.append(server.name)
+            else:
+                # We only need to collect the server names
+                servers.append(server.name)
+
+        for i in range(len(servers)):
+            # Set the values which will send this email(s)
+            if receiver != "":
```

Comment:
So if I specify a list to send the report to, it will only receive one message. But if I do not and user "Alice" has N > 1 server, she will still receive N emails.

Why don't you change this to an associate array. Then the above loop can just grab the recipient email (from either the `receiver` value or the `user.email` value), and append the value to a list. Then this can just iterate the keys of the array and send out one email to each address.

Something like this, starting at line 155 above:
```python
servers = {}
for server in self.__targets:
    user = conn.get_user_by_id(server.user_id, False)
    recipient = receiver if receiver != "" else user.email
    if recipient in servers:
        servers[recipient]["servers"].append(server.name)
    else:
        servers[recipient] = {"servers": [server.name], "name": user.name}

for email in servers:
    server_names = "\n".join(servers[email]["servers"])
    user = servers[email]["name"]
```
    etc